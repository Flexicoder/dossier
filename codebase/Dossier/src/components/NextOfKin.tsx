import React, { Component } from 'react';
import { Text, TextInput, View } from 'react-native';
import  styles  from '../assets/styles'

interface NextOfKinProps {
    name?: string | null;
    relationship?: string | null;
    address?: string | null;
    contactNumber?: string | null;
    email?: string | null;
}

export default class NextOfKin extends Component<NextOfKinProps> {
    constructor(props: NextOfKinProps) {
        super(props)
        this.state = { name: props.name,
                       relationship: props.relationship,
                       address: props.address,
                       contactNumber: props.contactNumber,
                       email: props.email }
    }

    render() {
        return (
        <View style={styles.container}>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Name"
                            onChangeText={text => this.setState({ name: text})}
                            defaultValue={this.props.name || ""} />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Relationship"
                            onChangeText={text => this.setState( { relationship: text })}
                            defaultValue={this.props.relationship || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Address"
                            onChangeText={text => this.setState( { address: text })}
                            defaultValue={this.props.address || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Contact Number"
                            onChangeText={text => this.setState( { contactNumber: text })}
                            defaultValue={this.props.contactNumber || ""}
                            />
                    </View>
                </View>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Email"
                            onChangeText={text => this.setState( { email: text })}
                            defaultValue={this.props.email || ""}
                            />
                    </View>
                </View>
            </View>
        );
    }
}