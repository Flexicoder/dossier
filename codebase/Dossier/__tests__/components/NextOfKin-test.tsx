
import 'react-native';
import React from 'react';
import NextOfKin from '../../src/components/NextOfKin';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("Key Information", () => {
    it('renders correctly', () => {
        const nextOfKin = renderer.create(<NextOfKin />);
        expect(nextOfKin.toJSON()).toMatchSnapshot();
      });
})


describe("Key Information", () => {
    it('contains supplied data', () => {
        const nextOfKin = renderer.create(<NextOfKin name="Bob" relationship="Father" />);
        const stringVersion = JSON.stringify(nextOfKin.toJSON())
        expect(stringVersion).toMatch(/Bob/);
        expect(stringVersion).toMatch(/Father/);
      });
})