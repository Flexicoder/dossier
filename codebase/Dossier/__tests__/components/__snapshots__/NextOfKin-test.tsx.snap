// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key Information renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Name"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Relationship"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Address"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Contact Number"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
